internal/manifest/testdata/oci1index.json

```diff
@@ -22,6 +22,16 @@
           "sse4"
         ]
       }
+    },
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "artifactType": "application/vnd.unknown.artifact.v1",
+      "size": 8221,
+      "digest": "sha256:615e8db5ae085b39e9cd24e5bc887d03fbd30ee4f55f5fedf9b697fafea4fbdd",
```

Comment:
Is this new entry used for anything? To exercise the match logic, to ensure the last entry is not chosen if there are similarly-matching alternatives preceding it, perhaps?

Reply:
Not originally, but it's about to start being used to ensure that we preserve the `artifactType` when editing a list that has an entry that already has a value.